src/client/common/process/processFactory.ts

```diff
@@ -16,9 +17,14 @@ export class ProcessServiceFactory implements IProcessServiceFactory {
         @inject(IEnvironmentVariablesProvider) private readonly envVarsService: IEnvironmentVariablesProvider,
         @inject(IProcessLogger) private readonly processLogger: IProcessLogger,
         @inject(IBufferDecoder) private readonly decoder: IBufferDecoder,
-        @inject(IDisposableRegistry) private readonly disposableRegistry: IDisposableRegistry
+        @inject(IDisposableRegistry) private readonly disposableRegistry: IDisposableRegistry,
+        @inject(IWorkspaceService) private readonly workspace: IWorkspaceService
     ) {}
     public async create(resource?: Uri): Promise<IProcessService> {
+        // This should never happen, but if it does ensure we never run code accidentally in untrusted workspaces.
+        if (!this.workspace.isTrusted) {
+            throw new Error('Workspace not trusted');
```

Comment:
Normally I would say this looks like it might end up user visible, so it would need to be localized. If the comment here is correct (we should never get here) ok as is, but if we think this might show up we should localize it.

Reply:
> f the comment here is correct (we should never get here) ok as is,

Yes, thats how it should work. Even if it doesn't work as expected, then we should fix the root cause instead of updating this message.